sql/sql_connect.cc

```diff
@@ -1000,6 +1000,28 @@ static int check_connection(THD *thd)
     bzero((char*) &net->vio->remote, sizeof(net->vio->remote));
   }
   vio_keepalive(net->vio, TRUE);
+
+  if (global_system_variables.keepalive_probes != 9)
```

Comment:
Thanks, this looks good to me. A minor suggestion would be to also  move the defaults outside of sql_const.h into some of the vio headers , and have VIO in the constant names ( VIO_DEFAULT_KEEAPALIVE_INTERVAL or something like that)